src/libs/antares/study/binding_constraint/BindingConstraintsList.cpp

```diff
@@ -0,0 +1,387 @@
+    // Copy of the name
+    ConstraintName name;
+    name = newname;
+    if (name == bc->name())
+        return true;
+    ConstraintName id;
+    Antares::TransformNameIntoID(name, id);
+    if (std::any_of(pList.begin(), pList.end(), [&id](auto constraint) {
+            return constraint->id() == id;
+        }))
+    {
+        return false;
+    }
+    bc->name(name);
+    JIT::Invalidate(bc->TimeSeries().jit);
+    return true;
+}
+
+bool BindingConstraintsList::loadFromFolder(Study &study,
+                                            const StudyLoadOptions &options,
+                                            const AnyString &folder) {
+    // Log entries
+    logs.info(); // space for beauty
+    logs.info() << "Loading constraints...";
+
+    // Cleaning
+    clear();
+
+    if (study.usedByTheSolver) {
+        if (options.ignoreConstraints) {
+            logs.info() << "  The constraints have been disabled by the user";
+            return true;
+        }
+        if (!study.parameters.include.constraints) {
+            logs.info() << "  The constraints shall be ignored due to the optimization preferences";
+            return true;
+        }
+    }
+
+    EnvForLoading env(study.areas, study.header.version);
+    env.folder = folder;
+
+    env.iniFilename << env.folder << Yuni::IO::Separator << "bindingconstraints.ini";
+    IniFile ini;
+    if (!ini.open(env.iniFilename)) {
+        return false;
+    }
+
+    // For each section
+    if (ini.firstSection) {
+        for (env.section = ini.firstSection; env.section; env.section = env.section->next) {
+            if (env.section->firstProperty) {
+               auto new_bc = LoadBindingConstraint(env);
+                std::copy(new_bc.begin(), new_bc.end(), std::back_inserter(pList));
+            }
+        }
+    }
+
+    bool hasError = checkTimeSeriesWidthConsistency();
+
+    // Logs
+    if (pList.empty())
+        logs.info() << "No binding constraint found";
+    else
+    {
+        std::sort(pList.begin(), pList.end(), compareConstraints);
+
+        if (pList.size() == 1)
+            logs.info() << "1 binding constraint found";
+        else
+            logs.info() << pList.size() << " binding constraints found";
+    }
+
+    // When ran from the solver and if the simplex is in `weekly` mode,
+    // all weekly constraints will become daily ones.
+    if (study.usedByTheSolver && sorDay == study.parameters.simplexOptimizationRange) {
+        mutateWeeklyConstraintsIntoDailyOnes();
+    }
+
+    InitializeTSNumbers();
+
+    return !hasError;
+}
+
+    bool BindingConstraintsList::checkTimeSeriesWidthConsistency() const {
+        bool hasError = false;
+        std::map<std::string, unsigned, std::less<>> timeSeriesCountByGroup;
+        for(const auto& bc: this->pList) {
+            auto count = timeSeriesCountByGroup[bc->group()];
+            auto width = bc->TimeSeries().width;
+            if (count == 0) {
```

Comment:
Some BCs may have a single time-series, this input should be considered valid.
```suggestion
            auto width = bc->TimeSeries().width;
            if (width == 1) continue;
            if (count == 0) {
```